### DISCUSSION TITLE
Enhancing the XCP Ecosystem: Towards a Stakeholder-Driven Network

### DISCUSSION THREAD

**Discussion by keyuno** [CounterpartyXCP/CIPs#129] [General]
**Introduction:**
This discussion seeks to propel the XCP ecosystem into a new era—one that prioritizes longevity and prosperity of the entire ecosystem. Beyond transactional interactions, the aim is to foster a deeper connection between participants and the protocol. The fundamental concept of having a stake in the network represents a collective commitment to the ongoing success of the protocol.

The genesis of XCP through the burning of Bitcoin serves to firmly establish its identity as an extension of the Bitcoin network. One of the primary purposes of XCP is to function as an anti-spam mechanism, leveraging its unique creation process to deter unwanted and frivolous activities within the ecosystem.

**Spam Prevention:**
The challenge of spam persists within the XCP ecosystem, creating clutter, inefficiency, and threats to overall system reliability. As our community expands, the potential for malicious actors to exploit the network grows, leading to increased congestion and disruptions.

**Having a Stake in the Protocol:**
Ensuring that participants have a stake in the protocol is essential for fostering responsibility and commitment within the XCP community. The absence of a tangible cost associated with some actions on the network may result in a lack of accountability among users. 

**Proposed Solution:**
To address these challenges, I propose the introduction of XCP fee **on every meaningful Counterparty action,** implemented through a proof-of-burn mechanism. This approach acts not only as a deterrent against spam but also ensures that participants have a financial stake in the protocol, fostering a more engaged and responsible community.

**Gas Analogy Concerns:** 
Some community members may express concerns that the proposed proof-of-burn mechanism, while akin to gas in traditional systems, may not inherently lead to stakeholding behavior. 

Unlike traditional staking models, burning XCP may lack the ongoing relationship and continuous benefits associated with stakeholding. This analogy raises questions about whether the proposed mechanism truly establishes a meaningful stake in the network.

**Addressing Gas Analogy Concerns:**
To address these concerns, it's crucial to highlight the deflationary nature and the impact of a lower token supply. Reducing the circulating supply of XCP, inherently contributes to a deflationary environment. This scarcity can serve as a strong incentive for participants to hold onto their XCP and expend it only on actions that hold meaningful value within the ecosystem.

**Cost Efficiency Concerns:** 
An additional concern arises as users are currently accustomed to paying fees in Bitcoin, questioning the practicality and perceived value of incorporating an XCP fee for rest of Counterparty actions. 

**Addressing Cost Efficiency Concerns:**
Acknowledging the concern around cost efficiency, particularly when users are accustomed to paying fees in Bitcoin, it is essential to delve into a nuanced discussion. The central focus of this discourse lies in determining the optimal balance in transaction costs. 

This balance should effectively deter spam while ensuring that the introduction of an XCP fee for each Counterparty action brings tangible benefits and perceived value to users. The ongoing conversation within the community will be pivotal in gauging the appropriate expenditure that aligns with the overarching objectives of promoting responsible usage and sustaining the prosperity of the XCP ecosystem.

**Added Friction Concerns :**
The introduction of additional friction to actions that did not previously require XCP, especially in creating numeric assets, may discourage participation in the network. This potential barrier raises concerns about the impact on user engagement and involvement within the ecosystem.

**Addressing Added Friction Concerns:**
Clear communication can help users understand the importance of certain actions and the positive impact on the overall health of the network.

**Community Involvement:**
This proposal is open for community discussion and feedback. Your thoughts, concerns, and suggestions are indispensable in shaping the future of the XCP ecosystem. 

**Further Reading:**
[Why Proof-of-Burn](https://counterparty.io/news/why-proof-of-burn/
)
[XCP as Anti-Spam Mechanism](https://github.com/CounterpartyXCP/Documentation/blob/master/Developers/protocol_specification.md)

**Comment by 0cam** [comment]
The perception of XCP is an issue. I perceive XCP as BTC recognized by the counterparty protocol. Many people perceive XCP as a pump and dump token. To help shift the focus away from the fact that there's a counterparty native currency and toward the benefits of the counterparty ecosystem I think XCP costs and payments should be translated to BTC and done behind the scenes by the wallet. Other CIPs suggest this and I am in favor of this. Ignorance is bliss, and I think this process would go a long way to help perception by shortening the metaphorical distance between XCP and BTC.

I also think XCP should be required for all operations in the protocol. If you want to interact with the BTC ledger you must pay a BTC fee. I don't see why this wouldn't hold true for counterparty. It drives all users to be stakeholders. Though I will say that again the legwork of obtaining XCP would be best done by the wallet or UI behind the scenes and relayed to the user as an additional BTC cost wrapped into the BTC txn fee.

XCP costs for various actions are another thing. They'd have to be dynamic in the instances I've mentioned and not at all inhibitive.

**Reply by 0cam** [reply]
To add to this, requiring XCP would drive the necessity of XCP and therefore it's value. One of the best things about counterparty is that it has a dex. The issue is XCP doesn't have much value so there's not as much liquidity as there could be. I think requiring XCP would drive its value, with the ultimate goal being creating liquidity in the asset/XCP markets to facility trustless exchanging of value. Otherwise the next option is BTCpay or dispensers.

**Reply by keyuno** [reply]
> The perception of XCP is an issue. I perceive XCP as BTC recognized by the counterparty protocol. Many people perceive XCP as a pump and dump token. To help shift the focus away from the fact that there's a counterparty native currency and toward the benefits of the counterparty ecosystem I think XCP costs and payments should be translated to BTC and done behind the scenes by the wallet. Other CIPs suggest this and I am in favor of this. Ignorance is bliss, and I think this process would go a long way to help perception by shortening the metaphorical distance between XCP and BTC.

Utilizing XCP should be conducted with transparency rather than being concealed, as it could be perceived as a deceptive practice. Users ought to be informed about the tokens they are spending, promoting a clear understanding of the technology. Educating users on the advantages and features of the Counterparty ecosystem, without resorting to hidden processes, is crucial for building trust and fostering a positive perception of XCP. 

> To add to this, requiring XCP would drive the necessity of XCP and therefore it's value. 

Requiring the use of XCP would indeed underscore its necessity, consequently influencing its overall value. It's essential to recognize that some may argue that holding assets within the protocol signifies a more significant stake than possessing XCP alone. 

This perspective is flawed because asset owners primarily prioritize the success of their specific assets and projects, while XCP serves as the token representing the overarching protocol. It's important to highlight that new token schemes that leverage Counterparty may lead to potential forks. Integrating XCP more deeply into the process serves as a measure to mitigate such risks.

**Reply by 0cam** [reply]
Transparency is fine too. The process of obtaining can be handled by the wallet. Before the transaction there can be an explanation that the wallet is purchasing XCP at the cheapest price and it can also say that XCP is the native token of counterparty. To design for the highest percentile the whole experience needs to be easy, yet it can also be transparent and informative.

I agree with you on the value of XCP. It creates a unity among users of the protocol and presents a hurtle to forks. This has played out to a lesser degree with stamps. Stamps just numerical assets on counterparty, yet the community naturally split into two and both sides view their asset community as an "us vs them". In reality its all counterparty users. Counterparty devs are struggling to define community consensus, but as they work on this the community grows less stable as it grows in population. Maybe XCP isn't the answer for this. But its all I can think of at the moment. Everyone succeeds if counterparty succeeds.

**Comment by jotapea** [comment]
I lean into the opposite direction, reduce the XCP dependence as much as possible. BUT named assets, what I like to refer as "superassets" (because of their capacity to create childern subassets), should ALWAYS require the XCP burn. Is a natural continuation of the BTC burned for XCP.

I would then remove the need of XCP burn for subassets, because these are not "superassets", and are numeric at the core.

**Reply by keyuno** [reply]
> I lean into the opposite direction, reduce the XCP dependence as much as possible. BUT named assets, what I like to refer as "superassets" (because of their capacity to create childern subassets), should ALWAYS require the XCP burn. Is a natural continuation of the BTC burned for XCP.

Token schemes leveraging Counterparty, particularly through Broadcasting and Numerics, during their initial launch might present issues. There's a potential threat of the people running these schemes turning hostile, thereby negatively impacting the ecosystem, including crucial elements such as block explorers, wallets, websites, and communication channels. Neglecting **deep integration** with XCP on significant actions introduces real, not just theoretical, risks that could compromise social consensus. It is crucial for every participant to have a stake in tokens representing the entire protocol. I hope you reconsider your position.

**Comment by keyuno** [comment]
Conversations regarding this form of consensus change ought to be documented here on GitHub rather than within Telegram groups.

**Counterparty users are XCP holders.** Implementing deep protocol integration with XCP, where burning XCP is required for any significant action on Counterparty, is highly logical as it serves to differentiate between genuine users and potential abusers.

Owning a stake in the XCP token signifies contribution to the entire protocol. Those who desire Counterparty's success demonstrate their commitment by investing according to their capabilities, recognizing that the protocol's success will enhance the value of their stake. Therefore, they voluntarily contribute to the ecosystem in whatever capacity they can, aligning their actions with their confidence in Counterparty's potential for growth.

XCP growth involves drawing increased attention, users, and developers, nurturing a more robust ecosystem—all without implementing taxes directed towards centralized developer funds with their accompanying liabilities. Genuine Counterparty users, who also hold XCP, are driven by their aspiration to witness Counterparty's prosperity. They actively participate in constructing applications, providing services, and bolstering infrastructure not in pursuit of direct fees obtained from fellow protocol participants, but because of their voluntary investment in the protocol and dedication to its success.

**Comment by Jpja** [comment]
I'm in favor of using **_XCP fees only when necessary_**, ie how it has always been. For most transactions the bitcoin fee is sufficient. For mass-transactions like dividend, sweep, multi-send, and certain dispenser cases, an XCP fee is necessary. The second case is squatting protection for the asset namespace.

Future cases for XCP fees are smart contracts and [bundled transactions](https://github.com/CounterpartyXCP/Forum/discussions/127).

Currently fees are static and far from perfect. Several spam vectors exist due to lacking XCP fees. Closing these should be a high priority. And eventually moving to a dynamic model would be great.
